src/charm.py

```diff
@@ -1131,12 +1133,13 @@ def _on_image_relation_changed(self, _: ops.RelationChangedEvent) -> None:
         state = self._setup_state()
 
         if state.instance_type != InstanceType.OPENSTACK:
+            self.unit.status = BlockedStatus("LXD mode enabled. Please remove image relation.")
```

Comment:
Not a big deal, but I personally would prefer something like "OpenStack mode not enabled. Please remove image relation".

The reason being this description is more future proof, and would not be outdated by new addition of modes.
Since if we do add new mode we might miss this error message during the refactor.